src/test/java/web/tests/FlipkartTest.java

```diff
@@ -168,19 +170,35 @@ public void testAddToCartFunctionality(String product) {
     }
 
     /**
-     *
-     * @param item :accepting shoes from json
+     * @param item     :accepting shoes from json
      * @param position :accepting position i.e to select 2nd position shoes
      */
     @ParameterizedTest
     @CsvFileSource(resources = "/testProductPageDesign.csv")
-    public void testProductPageDesign(String item,int position){
+    public void testProductPageDesign(String item, int position) {
         ProductPage productPage = new HomePage().popUpCancel().setShoes(item).searchShoes().OpenProductPage(position);
         productPage.scrollToBottom();
-        ArrayList<String> headers  = new ArrayList<>(Arrays.asList(map.get("headers").split(",")));
-        for(String headersInSelectionPage:headers){
+        ArrayList<String> headers = new ArrayList<>(Arrays.asList(map.get("headers").split(",")));
+        for (String headersInSelectionPage : headers) {
             softAssert.assertThat(productPage.isSelectionDisplayed(headersInSelectionPage)).
                     as("Json selected Headers is Not matching with selection page Headers").isTrue();
         }
     }
+
+    @Test
+    public void flipkartSocialMedia() {
+        HomePage homePage = new HomePage().popUpCancel();
+        ArrayList<String> socialMediaLinks = new ArrayList<>(Arrays.asList(map.get("links")
+                .split(",")));
+        for (int i = 0; i < socialMediaLinks.size(); i++) {
+            SocialMediaPage socialMediaPage =homePage.clickLink(socialMediaLinks.get(i));
+            sleep(4000);
+            String url = socialMediaPage.getSocialMediaUrl();
+            if (!url.contains(socialMediaLinks.get(i).toLowerCase())) {
+                softAssert.assertThat((socialMediaLinks.get(i))).
+                        as(" Json selected Links contents is Not matching with socialMedia page url");
```

Comment:
incorrect validation : 
1. Why are we using an if statement?
2. soft assertion is asserting anything it seems. Please verify the syntax/logic